Source/Foundation/bsfUtility/Prerequisites/BsStdHeaders.h

```diff
@@ -223,13 +223,18 @@ namespace bs
 		constexpr T& operator*() const { return *mPtr; }
 		constexpr T* operator->() const { return mPtr; }
 		constexpr T* get() const { return mPtr; }
+		constexpr bool operator<(const NativePtr& rhs) const noexcept {
+			constexpr std::less<> cmp = {};
```

Comment:
What was breaking because this operator wasn't defined?

Also, why define cmp? Why not simply do

return mPtr < rhs.mPtr; ?

They're raw pointers, so there's no way for that comparison to be ambiguous, is there?

Reply:
Take a look at my question on [reddit](https://www.reddit.com/r/cpp_questions/comments/8hxfuq/problems_with_gcc_81_stdset_and_stdless/)

Reply:
That's really surprising that GCC would assert on this.

Oh well. They do weird things, it's no big deal.

We could still simplify the guts of the function though, couldn't we?

Reply:
While we're at it, I suppose we might as well define all the operators, just so it's not a problem in the future.

Want me to do that?

Reply:
Agreed. Will have to do that in another PR since this one was merged ?

Reply:
Alternatively, I didn't try this before, but it might work to instead define an implicit conversion operator to T*.

Reply:
Yea, it would have to be in a different PR.

If you want to make the PR, please feel free. Or you can punt it to me and I'll do it tonight

Reply:
Be my guest. I'm trying to do a few clang/ubsan fixes right now :smile: